phoenix-core/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -156,7 +157,7 @@ private static MutationState upsertSelect(PhoenixStatement statement,
                             column.getMaxLength(), column.getScale(), column.getSortOrder());
                     values[i] = ByteUtil.copyKeyBytesIfNecessary(ptr);
                 }
-                setValues(values, pkSlotIndexes, columnIndexes, table, mutation);
+                setValues(values, pkSlotIndexes, columnIndexes, table, mutation, statement);
```

Comment:
It should be OK because I have separated incrementing and getting the statement index. Incrementing should only be done by the original statement/connection. Cloned connections should only be getting the statement index. I think I need to do more testing here, though. Thanks.